cherab/solps/solps_plasma.py

```diff
@@ -17,228 +17,577 @@
+        _check_shape("velocities_cylindrical", value, (len(self.species_list), 3, self.mesh.ny, self.mesh.nx))
+
+        # Converting to poloidal coordinates
+        velocities = np.zeros(value.shape)
+        velocities[:, 2] = -value[:, 1]
+        for k in range(value.shape[0]):
+            velocities[k, :2] = self.mesh.to_poloidal(value[k, [0, 2]])
+
+        self._velocities_cylindrical = value
+        self._velocities = velocities
+        self._velocities_cylindrical_f2d = {}
+        self._velocities_cartesian = {}
+        for k, sp in enumerate(self._species_list):
+            self._velocities_cylindrical_f2d[k] = SOLPSVectorFunction2D.instance(self._sample_vector_f2d, value[k])
+            self._velocities_cylindrical_f2d[sp] = self._velocities_cylindrical_f2d[k]
+            self._velocities_cartesian[k] = VectorAxisymmetricMapper(self._velocities_cylindrical_f2d[k])
+            self._velocities_cartesian[sp] = self._velocities_cartesian[k]
+
+    @property
+    def velocities_cylindrical_f2d(self):
+        """
+        Dictionary of VectorFunction2D interpolators for species velocities
+        in cylindrical coordinates.
+        Accessed by species index or species_list elements.
+        E.g., elocities_cylindrical_f2d[1] or elocities_cylindrical_f2d[('deuterium', 1))].
+        Each entry returns a vector of species velocity at a given point (R, Z).
+        :return:
+        """
+        return self._velocities_cylindrical_f2d
 
     @property
     def velocities_cartesian(self):
+        """
+        Dictionary of VectorFunction3D interpolators for species velocities
+        in Cartesian coordinates.
+        Accessed by species index or species_list elements.
+        E.g., elocities_cylindrical_f3d[1] or elocities_cylindrical_f3d[('deuterium', 1))].
+        Each entry returns a vector of species velocity at a given point (x, y, z).
+        :return:
+        """
         return self._velocities_cartesian
 
+    @property
+    def inside_mesh(self):
+        """
+        Function2D for testing if point p is inside the simulation mesh.
+        """
+        return self._inside_mesh
+
     @property
     def inside_volume_mesh(self):
         """
         Function3D for testing if point p is inside the simulation mesh.
         """
-        if self._inside_mesh is None:
-            raise RuntimeError("Inside mesh test not available for this simulation")
-        else:
-            return self._inside_mesh
+        return AxisymmetricMapper(self._inside_mesh)
 
     @property
     def total_radiation(self):
         """
-        Total radiation
+        Total radiation at each mesh cell.
+        Array of shape (ny, nx).
 
         This is not calculated from the CHERAB emission models, instead it comes from the SOLPS output data.
         Is calculated from the sum of all integrated line emission and all Bremmstrahlung. The signals used are 'RQRAD'
         and 'RQBRM'. Final output is in W/str?
```

Comment:
I think this should be in W/m3, based on #28.

Reply:
I corrected this to W m-3, however this docstring is valid only if `total_radiation` is read from MDSPlus server. When obtained from raw output, `total_radiation` includes only radiation from neutral atoms, when obtained from balance.nc, it does not include bremsstrahlung.